### PR TITLE
22 pkmn controller endpoint

### PR DESCRIPTION
**Pull request by SanderMeister**
if we need more endpoints please let me know

### CODE REVIEW

**Review by SimonBaars — APPROVED** [review]

File: src/main/kotlin/resource/PokemonResource.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.ktor.routing.route
+import service.pokemon.PokemonService
+import utility.PokeApi
+import utility.PokeApiAdapter
+
+fun Route.pokemon(pokemonService: PokemonService) {
+    route("/pokemon") {
+        get("/{id}") {
+            val id = call.parameters["id"]
+            if (id != null) {
```
Comment (SimonBaars) [inline src/main/kotlin/resource/PokemonResource.kt]: Not needed

File: src/main/kotlin/resource/PokemonResource.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                call.respond(PokeApiAdapter.getPokemonData(id.toInt()))
+            }
+        }
+
+        get("/pokedex/{id}") {
+            val id = call.parameters["id"]
+            if (id != null) {
```
Comment (SimonBaars) [inline src/main/kotlin/resource/PokemonResource.kt]: Not needed?

File: src/main/kotlin/resource/PokemonResource.kt (hunk 1 above)
Comment (SanderMeister) [reply src/main/kotlin/resource/PokemonResource.kt]: Isn't it? How would you implement it?

Comment (loehnertz) [reply src/main/kotlin/resource/PokemonResource.kt]: It turned out that `ktor` will not even call this endpoint handler when the query does not match.
Meaning that if no `id` is supplied, the handler will never be called and a `404` is directly returned instead as no handler matched the request. That's why we don't need to do this nullability check there.

You could not know though of course :)

**Review by loehnertz — APPROVED** [review]
I removed the nullability checks and ran a format on the code base.

**Pull request merged**